### PR TITLE
Improve HasTraits introspection with `dir()`

### LINKED ISSUE #925: HasTraits._notifiers method not showing up in dir()
The `HasTraits.__dir__` implementation appears to suppress names of some methods that would otherwise show up. An example is the `_notifiers` method. For introspection purposes, it would be very useful to have those method names appear as normal.

### PR DESCRIPTION
**Pull request by midhun-pm**
Fix #925 
Includes the names from `super().__dir__()` in the `HasTraits.__dir__`'s output.

### CODE REVIEW

File: traits/has_traits.py [hunk 1]
```diff
@@ -1355,7 +1355,8 @@ def __dir__(self):
         trait_names = self.trait_names()
         method_names = [method for method in self._each_trait_method(self)]
         class_attrs = list(vars(self.__class__).keys())
-        return trait_names + method_names + class_attrs
+        return list(
+            set(trait_names + method_names + class_attrs + super().__dir__()))
```
Comment (corranwebster) [inline traits/has_traits.py]: I _think_ if you use `super` then you shouldn't need to  do `method_names` and `class_attrs`: they should be picked up by the standard implementation.

File: traits/tests/test_has_traits.py [hunk 2]
```diff
@@ -366,6 +366,18 @@ def test__trait_set_inited(self):
 
         self.assertTrue(foo.traits_inited())
 
+    def test___dir__(self):
```
Comment (corranwebster) [inline traits/tests/test_has_traits.py]: There is already a `test_dir` test in `test_get_traits.py`; that should probably be merged with this one (it tests that uninitialized traits and non-trait class variables are reported correctly).

File: traits/has_traits.py (hunk 1 above)
Comment (midhun-pm) [reply traits/has_traits.py]: Yes, you are right, verified that. Thanks.

**Comment by mdickinson** [comment]
Envisage seems fine: the Envisage test suite runs without error on Envisage master + Traits from this branch. Merging.

**Comment by mdickinson** [comment]
Aargh. Sorry, @corranwebster: I should have waited for you to confirm that you're happy that your comments have been addressed before merging. Let me know if not, and we can revert or do a follow-up PR as appropriate.

**Pull request merged**